src/gate_kernels.jl

```diff
@@ -481,8 +481,84 @@ apply_gate!(::Val{false}, g::I, state_vec::AbstractStateVector{T}, qubits::Int..
     return
 apply_gate!(::Val{true}, g::I, state_vec::AbstractStateVector{T}, qubits::Int...) where {T<:Complex} =
     return
-apply_gate!(::Measure, state_vec, args...) = return
-apply_gate!(::Reset, state_vec, args...) = return
+    
+"""
+    apply_gate!(measure_op::Measure, state_vec::AbstractStateVector{T}, qubit::Int) 
+                            where {T<:Complex}
+
+Applies a measurement projector into the subspace defined by the measurement outcome
+and target stored in the measure_op. Specifically applies the operation on a statevector.
+"""    
+
+function apply_gate!(measure_op::Measure, state_vec::AbstractStateVector{T}, qubit::Int) where {T<:Complex}
+    # If result is not set (default -1), just return without doing anything
+    measure_op.result == -1 && return
+    
+    n_amps, endian_qubit = get_amps_and_qubits(state_vec, qubit)
+    
+    # Create a mask for the qubit we're measuring
+    mask = 1 << endian_qubit
+    
+    # Iterate through all amplitudes
+    for i in 0:n_amps-1
+        # Check if the qubit is in state 0 or 1
+        qubit_value = (i & mask) >> endian_qubit
+        
+        # If the qubit value doesn't match the measurement result, set amplitude to 0
+        if qubit_value != measure_op.result
+            state_vec[i+1] = zero(T)
+        end
+    end
+    
+    # Normalize the state vector
+    norm_factor = 1.0 / sqrt(sum(abs2.(state_vec)))
+    if !isinf(norm_factor)
+        state_vec .*= norm_factor
+    end
+    
+    return
+end
+
+function apply_gate!(::Reset, state_vec::AbstractStateVector{T}, qubit::Int) where {T<:Complex}
```

Comment:
There's already an implementation for reset implemented in the simulator files; why have another here?